src/core/actions/cityactions/ResourceGathering.java

```diff
@@ -23,20 +24,69 @@ public ResourceGathering(City c)
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute resource gathering
-        return null;
+        LinkedList<Action> resources = new LinkedList<>();
+        if(isFeasible(gs)){
```

Comment:
This wouldn't work. The object ResourceGathering would not have "this.resource" set to any actual value when this function is being called. This function needs to create an action for each one of the resources available for this city (if there are enough stars to make the gather)